crypto/params.c

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2019, Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <openssl/bn.h>
+#include <openssl/params.h>
+#include <openssl/err.h>
+#include <openssl/cryptoerr.h>
+
+OSSL_PARAM *OSSL_PARAM_locate(const OSSL_PARAM *p, const char *key)
```

Comment:
There's no reason for it to return a non-const

Reply:
yes there is:  it is used in the "set" functions.

Reply:
set functions should not modify the `OSSL_PARAM` itself, only what it points at

Reply:
In the `_PTR` variant for strings, the `buffer` pointer is updated on set, unless I've missed something.

Reply:
Ignore that, I'd made a mistaken assumption about the _PTR.